src/qibocal/protocols/characterization/classification.py

```diff
@@ -34,16 +34,22 @@ def nshots(self):
 
 @dataclass
 class SingleShotClassificationResults(Results):
-    threshold: Dict[List[Tuple], str] = field(metadata=dict(update="threshold"))
-    rotation_angle: Dict[List[Tuple], str] = field(metadata=dict(update="iq_angle"))
-    mean_gnd_states: Dict[List[Tuple], str] = field(
+    """SingleShotClassification outputs."""
+
+    threshold: Dict[Union[str, int], float] = field(metadata=dict(update="threshold"))
+    """Threshold for classification."""
+    rotation_angle: Dict[Union[str, int], float] = field(
+        metadata=dict(update="iq_angle")
+    )
+    """Threshold for classification."""
+    mean_gnd_states: Dict[Union[str, int], complex] = field(
         metadata=dict(update="mean_gnd_states")
     )
-    mean_exc_states: Dict[List[Tuple], str] = field(
+    mean_exc_states: Dict[Union[str, int], complex] = field(
```

Comment:
I understand that this, and all other changes here would be replaced by
```suggestion
    mean_exc_states: Dict[QubitId, complex] = field(
```
Does it make sense to have
```suggestion
    mean_exc_states: Dict[Qubit, complex] = field(
```
instead? This way we would avoid creating a new type altogether. I undrestand this may not work as expected with the `MultiqubitPlatform` (but maybe it will), but it would be good to design these things with the future in mind.

Reply:
> Yes, I agree that defining the `QubitId` type in qibolab makes sense. It will probably make maintenance easier too, if in the future we support only strings (or only ints) we just change `QubitId = str` in one place in qibolab instead of having to update all routines here. If you want I can do it in [qiboteam/qibolab#387](https://github.com/qiboteam/qibolab/pull/387).

Sure, if you agree I believe that we can implement it.

I think that the second approach will not work.
If you try something like this:
```py
from qibolab.platforms.abstract import Qubit
qubit = Qubit(0)
test_dict = {qubit : "qubit"} 
```
It will raise the following error
```sh
TypeError: unhashable type: 'Qubit'
```
we could define `__hash__` for Qubit but I don't what is better between this and the QubitId type.

Reply:
Right, I guess `Qubit` is not hashable since it is mutable. We need it to be mutable as we typically change its parameters (frequencies, etc.) during calibration, otherwise we would need to create a new `Qubit` everytime. It is probably possible to make a mutable object hashable, but I am not sure if it is a good idea.

Maybe just defining the type `QubitId` is the best solution. I will do it there.